src/app/data-entries/input-group/reactive-form-custom-validation/reactive-form-custom-validation.component.ts

```diff
@@ -59,75 +60,66 @@ export class ReactiveFormCustomValidationComponent {
         this.registrationForm.reset();
     }
 
-    private usernameValidator(val: string): ValidatorFn {
+    private emailValidator(val: string): ValidatorFn {
         return (control: AbstractControl): ValidationErrors | null => {
-            if (control.value?.toLowerCase().includes(val)) {
-                return { username: true };
+            const value = control.value?.toLowerCase();
+            const localPartRegex = new RegExp(`(?<=(${val})).*[@]`);
+            const domainRegex = new RegExp(`(?<=[@])(?=.*(${val}))`);
+
+            const returnObj = {};
```

Comment:
Make an interface describing the error object and assign that type to the return object. Then you wouldn't have to use `['prop']` notation and it also looks cleaner!